### PR TITLE
add lease permissions for library-go update

### PR DESCRIPTION
**Pull request by skeeey**
Signed-off-by: Wei Liu <liuweixa@redhat.com>

### CODE REVIEW

File: manifests/klusterlet/managed/klusterlet-registration-clusterrole.yaml [hunk 1]
```diff
@@ -22,7 +22,7 @@ rules:
 - apiGroups: ["cluster.open-cluster-management.io"]
   resources: ["clusterclaims"]
   verbs: ["get", "list", "watch"]
-# Allow agent to list addons lease
+# Allow agent to operate leases
 - apiGroups: ["coordination.k8s.io"]
   resources: ["leases"]
-  verbs: ["get", "list", "watch"]
+  verbs: ["create", "get", "list", "watch", "update", "patch"]
```
Comment (qiujian16) [inline manifests/klusterlet/managed/klusterlet-registration-clusterrole.yaml]: these are agent rbac on hub I think, lease is not needed

File: manifests/klusterlet/managed/klusterlet-work-clusterrole.yaml [hunk 2]
```diff
@@ -33,3 +33,7 @@ rules:
 - apiGroups: ["work.open-cluster-management.io"]
   resources: ["appliedmanifestworks/finalizers"]
   verbs: ["update"]
+# Allow agent to operate leases
+- apiGroups: ["coordination.k8s.io"]
+  resources: ["leases"]
```
Comment (qiujian16) [inline manifests/klusterlet/managed/klusterlet-work-clusterrole.yaml]: these are agent rbac on hub I think, lease is not needed

File: manifests/klusterlet/managed/klusterlet-registration-clusterrole.yaml (hunk 1 above)
Comment (skeeey) [reply manifests/klusterlet/managed/klusterlet-registration-clusterrole.yaml]: this is the agent clusterrole on spoke, we already add the lease permission to the spoke agent role (`klusterlet/management/klusterlet-registration-role.yaml`) , so rollbacked

File: manifests/klusterlet/managed/klusterlet-work-clusterrole.yaml (hunk 2 above)
Comment (skeeey) [reply manifests/klusterlet/managed/klusterlet-work-clusterrole.yaml]: this is the agent clusterrole on spoke, we already add the lease permission to the spoke agent role (`klusterlet/management/klusterlet-work-role.yaml`) , so rollbacked

**Review by qiujian16 — COMMENTED** [review]
/lgtm

**Pull request merged**